apps/challenges/views.py

```diff
@@ -2973,6 +2974,81 @@ def get_worker_logs(request, challenge_pk):
     return Response(response_data, status=status.HTTP_200_OK)
 
 
+@api_view(["PUT"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
+def scale_resources_by_challenge_pk(request, challenge_pk):
+    """
+    The function called by a host to update the resources used by their challenge.
+
+    Calls the scale_resources method. Before calling, checks if the caller hosts the challenge and provided valid CPU
+    unit counts and memory sizes (MB).
+
+    Arguments:
+        request {HttpRequest} -- The request object
+        challenge_pk {int} -- The challenge pk for which its workers' resources will be updated
+
+    Returns:
+        Response object -- Response object with appropriate response code (200/400/403/404)
+    """
+    if not is_user_a_host_of_challenge(request.user, challenge_pk):
+        response_data = {
+            "error": "Sorry, you are not authorized for access worker operations."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if request.data.get("worker_cpu_cores") is None:
+        response_data = {
+            "error": "CPU unit count config missing from request data."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if request.data.get("worker_memory") is None:
+        response_data = {
+            "error": "Worker memory config missing from request data."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    challenge = get_challenge_model(challenge_pk)
+    cpu_cores = request.data["worker_cpu_cores"]
+    memory = request.data["worker_memory"]
+
+    try:
+        cpu_cores = int(cpu_cores)
+        memory = int(memory)
+    except Exception:
+        response_data = {
+            "error": "Enter integer numbers for cores and memory."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
You can just parse it in int on L3014 and L3015. We don't need exception handling here. If the type is incorrect API will throw a 500

Reply:
Good to know, thanks!